### DISCUSSION TITLE
Add Padding via dragging component

### DISCUSSION THREAD

**Discussion by Heapware** [GrapesJS/grapesjs#4101] [Q&A]
Hello Team Grapesjs & @artf 
This Is Haris Ali, I have been using grapesjs for the last 8 months, I was facing an issue that I want to add padding via dragging the component, but unable to do that. Like if I add padding via dragging component then the component becomes non-editable and if I make it editable the padding functionality via dragging doesn't work. 
Could you please help me to resolve this issue?
looking forward to hearing from you. 

Thanks

**Comment by artf** [comment]
Sorry, but what you're asking sounds like no sense. Try to explain better, show some code...

**Reply by Heapware** [reply]
Hello @artf 
Let me share a video of this, I hope you will understand better. Please check this video and let me know, 
Thanks & Regards
Haris Ali

https://user-images.githubusercontent.com/86477536/151160441-ba6817ab-aca4-476a-adb4-aecafee9b62e.MP4

**Reply by artf** [reply]
Ah ok, I got it. For now, the core doesn't provide such functionality but unfortunately even implementing via API wouldn't be that easy. I'd say to wait for the canvas refactoring where I'm planning to support selected component rect customization, then such a thing will be easier to implement.

**Reply by Heapware** [reply]
Hey @artf 
Could you please tell me how I can do this via some API, and which API to use to achieve this functionality? I have created a complete website builder with this even with custom styling. Just tell me from which API I can achieve this I will do my R&D too. 
Please share some of the information so I can start my research I hope I will find a way to implement that API. 
Thank you
